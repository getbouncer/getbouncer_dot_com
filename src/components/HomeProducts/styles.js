import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const SectionContainer = styled.section`
  margin: 150px 0 80px;

  @media ${breakpoints.smartphone} {
    margin: 115px 0 80px;
  }
`

export const Headline = styled.div`
  margin-bottom: 64px;
  text-align: center;

  @media ${breakpoints.smartphone} {
    margin-bottom: 50px;
    text-align: left;
  }

  h2 {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    display: inline-block;
    font-size: 2.857rem;
  }
`

export const Item = styled.li`
  align-items: center;
  display: flex;

  @media ${breakpoints.smartphone} {
    flex-wrap: wrap;
  }

  &:not(:last-child) {
    margin-bottom: 60px;
  }

  &:nth-of-type(odd) {
    @media ${breakpoints.smartphone} {
      justify-content: center;
      overflow: hidden;
    }


    > img {
      margin-right: 15%;

      @media ${breakpoints.smartphone} {
        margin-right: 0px;
        order: 2;
      }
    }
  }

  &:nth-of-type(even) {
    justify-content: flex-end;

    @media ${breakpoints.smartphone} {
      justify-content: center;
      overflow: hidden;
    }

    > img {
      margin-left: 5%;
      order: 2;

      @media ${breakpoints.smartphone} {
        margin-left: 0px;
        margin-top: 42px;
      }
    }
  }
`

export const ItemImage = styled.img`
  width: 50%;

  @media ${breakpoints.smartphone} {
    max-width: 550px;
    width: 100%;
  }
`

export const ItemIcon = styled.img`
  display: block;
  height: 32px;
`

export const ItemHeadline = styled.div`
  margin: 15px 0;

  @media ${breakpoints.smartphone} {
    margin: 15px 0 32px;
  }

  h3 {
    font-size: 2.2rem;
  }
`

export const ItemText = styled.div`
  margin-bottom: 20px;

  @media ${breakpoints.smartphone} {
    margin-bottom: 36px;
  }
`

export const ItemContent = styled.div`
  width: 30%;

  @media ${breakpoints.smartphone} {
    width: 100%;
  }
`
