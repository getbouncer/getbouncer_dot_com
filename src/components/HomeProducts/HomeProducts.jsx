import React from 'react'
import PropTypes from 'prop-types'
import Container from '~globals/Container'
import CtaButton from '~components/CtaButton'
import { SectionContainer, Headline, Item, ItemImage, ItemIcon, ItemContent, ItemHeadline, ItemText } from './styles'

const HomeProducts = ({ input }) => {
  const productNodes = input.items

  return (
    <SectionContainer>
      <Container>
        <Headline dangerouslySetInnerHTML={{ __html: input.primary.products_headline.html }} />
        <ul>
          {productNodes.map(product => {
            const productItem = product.products.document.data
            return (
              <Item key={product.products.document.uid}>
                <ItemImage src={productItem.product_image_home.url} alt={productItem.product_image_home.alt} />
                <ItemContent>
                  <ItemIcon src={productItem.product_logo.url} alt={productItem.product_logo.alt} />
                  <ItemHeadline dangerouslySetInnerHTML={{ __html: productItem.product_headline_home.html }} />
                  <ItemText dangerouslySetInnerHTML={{ __html: productItem.product_description_home.html }} />
                  <CtaButton text={productItem.product_cta_home} to={`/${product.products.document.uid}`} />
                </ItemContent>
              </Item>
            )
          })}
        </ul>
      </Container>
    </SectionContainer>
  )
}

HomeProducts.propTypes = {
  input: PropTypes.shape().isRequired,
}

export default HomeProducts
