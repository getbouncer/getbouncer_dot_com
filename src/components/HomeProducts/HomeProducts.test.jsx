import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import HomeProducts from './HomeProducts'

describe('HomeProduct', () => {
  it('renders correctly', () => {
    const data = {
      primary: {
        products_headline: {
          html: '<h1>products headline</h1>',
        },
      },
      items: [
        {
          products: {
            document: {
              data: {
                product_logo: {
                  alt: 'product 1 logo',
                  url: 'https://www.example.com/product_1_logo.png',
                },
                product_headline_home: {
                  html: '<h3>product 1 headline home</h3>',
                },
                product_description_home: {
                  html: '<p>product 1 description home</p>',
                },
                product_image_home: {
                  alt: 'product 1 image home',
                  url: 'https://www.example.com/product_1_image_home.png',
                },
                product_cta_home: 'product 1 cta home',
              },
              uid: 'product 1',
            },
          },
        },
        {
          products: {
            document: {
              data: {
                product_logo: {
                  alt: 'product 2 logo',
                  url: 'https://www.example.com/product_2_logo.png',
                },
                product_headline_home: {
                  html: '<h3>product 2 headline home</h3>',
                },
                product_description_home: {
                  html: '<p>product 2 description home</p>',
                },
                product_image_home: {
                  alt: 'product 2 image home',
                  url: 'https://www.example.com/product_2_image_home.png',
                },
                product_cta_home: 'product 2 cta home',
              },
              uid: 'product 2',
            },
          },
        },
      ],
    }
    const tree = renderer
      .create(<HomeProducts input={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
