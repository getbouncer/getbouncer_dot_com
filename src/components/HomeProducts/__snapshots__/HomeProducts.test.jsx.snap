// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeProduct renders correctly 1`] = `
<section
  className="css-4hxivr"
>
  <div
    className="css-1fywgw4"
  >
    <div
      className="css-1pp6h0w"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>products headline</h1>",
        }
      }
    />
    <ul>
      <li
        className="css-1n9onjb"
      >
        <img
          alt="product 1 image home"
          className="css-1uwazzl"
          src="https://www.example.com/product_1_image_home.png"
        />
        <div
          className="css-1ovmt8n"
        >
          <img
            alt="product 1 logo"
            className="css-dqfcdz"
            src="https://www.example.com/product_1_logo.png"
          />
          <div
            className="css-tef5ix"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<h3>product 1 headline home</h3>",
              }
            }
          />
          <div
            className="css-1to6z5u"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>product 1 description home</p>",
              }
            }
          />
          <a
            className="css-e1lh64"
            href="/product 1"
          >
            product 1 cta home
          </a>
        </div>
      </li>
      <li
        className="css-1n9onjb"
      >
        <img
          alt="product 2 image home"
          className="css-1uwazzl"
          src="https://www.example.com/product_2_image_home.png"
        />
        <div
          className="css-1ovmt8n"
        >
          <img
            alt="product 2 logo"
            className="css-dqfcdz"
            src="https://www.example.com/product_2_logo.png"
          />
          <div
            className="css-tef5ix"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<h3>product 2 headline home</h3>",
              }
            }
          />
          <div
            className="css-1to6z5u"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>product 2 description home</p>",
              }
            }
          />
          <a
            className="css-e1lh64"
            href="/product 2"
          >
            product 2 cta home
          </a>
        </div>
      </li>
    </ul>
  </div>
</section>
`;
